pynndescent/__init__.py

```diff
@@ -1,7 +1,14 @@
-import pkg_resources
+import sys
+
 import numba
+
 from .pynndescent_ import NNDescent, PyNNDescentTransformer
 
+if sys.version_info[:2] >= (3, 8):
+    import importlib.metadata as importlib_metadata
+else:
+    import importlib_metadata
+
```

Comment:
Another option here is to just catch the ImportError (and you wouldn't need to import `sys`)

```suggestion
try:
    import importlib.metadata as importlib_metadata
except ImportError:
    import importlib_metadata
```

Reply:
I'd prefer not to swallow any exceptions

Reply:
Fair enough! Suggested because it's a fairly common pattern (there's an `except ImportError` in this file already).

A totally different (over-the-top) solution to this issue would be to rewrite the packaging to use `setup.cfg` and `pyproject.toml` files, at which point `setup.py` is no longer necessary and the version can be stored as a literal in this file (`setup.cfg` can read it). I only learned how to do this yesterday, so I'm mad with power. :joy: